engine/src/main/java/org/terasology/world/block/family/FreeformFamily.java

```diff
@@ -16,27 +16,113 @@
 
 package org.terasology.world.block.family;
 
-import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.math.Rotation;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.naming.Name;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockBuilderHelper;
 import org.terasology.world.block.BlockUri;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
 
-import java.util.Collections;
-import java.util.List;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
 
 /**
  * A freeform family is a pseudo block family that can be combined with any block shape to produce an actual block
  * family.
- *
  */
-public class FreeformFamily {
-    public BlockUri uri;
-    public List<String> categories;
+@RegisterBlockFamily("freeform")
+@FreeFormSupported(true)
+public class FreeformFamily extends AbstractBlockFamily {
```

Comment:
Hah, this might be the best spot to better describe what a "freeform family" is - maybe we have some good examples?